pkg/upgrade/ready_test.go

```diff
@@ -0,0 +1,286 @@
+package upgrade
+
+import (
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func TestIsPodReady(t *testing.T) {
+	tests := []struct {
+		name string
+		pod  *corev1.Pod
+		want bool
+	}{
+		{
+			name: "ready",
+			want: true,
+			pod: &corev1.Pod{
+				Status: corev1.PodStatus{
+					Conditions: []corev1.PodCondition{
+						{
+							Type:   corev1.PodInitialized,
+							Status: corev1.ConditionTrue,
+						},
+						{
+							Type:   corev1.PodReady,
+							Status: corev1.ConditionTrue,
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "missing",
+			want: false,
+			pod: &corev1.Pod{
+				Status: corev1.PodStatus{
+					Conditions: []corev1.PodCondition{
+						{
+							Type:   corev1.PodInitialized,
+							Status: corev1.ConditionFalse,
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		if got := isPodReady(tt.pod); got != tt.want {
+			t.Errorf("isPodReady(%s) = %v, want %v", tt.name, got, tt.want)
+		}
+	}
+}
+
+func TestNodeReady(t *testing.T) {
+	tests := []struct {
+		name string
+		node *corev1.Node
+		want bool
+	}{
+		{
+			name: "ready",
+			want: true,
+			node: &corev1.Node{
+				Status: corev1.NodeStatus{
+					Conditions: []corev1.NodeCondition{
+						{
+							Type:   corev1.NodeOutOfDisk,
+							Status: corev1.ConditionFalse,
+						},
+						{
+							Type:   corev1.NodeReady,
+							Status: corev1.ConditionTrue,
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "missing",
+			want: false,
+			node: &corev1.Node{
+				Status: corev1.NodeStatus{
+					Conditions: []corev1.NodeCondition{
+						{
+							Type:   corev1.NodeOutOfDisk,
+							Status: corev1.ConditionFalse,
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		if got := isNodeReady(tt.node); got != tt.want {
+			t.Errorf("isNodeReady(%s) = %v, want %v", tt.name, got, tt.want)
+		}
+	}
+}
+
+func TestNodeIsReady(t *testing.T) {
+	tests := []struct {
+		name     string
+		kc       kubernetes.Interface
+		nodeName string
+		want     bool
+		wantErr  bool
```

Comment:
can you easily test that if the clientset returns some other error, it propagates?

Reply:
I have tried using AddReactor, but it doesn't seem to be hitting.